operator/cmd/mesh/profile-common.go

```diff
@@ -140,8 +156,41 @@ func genIOPS(inFilename []string, profile, setOverlayYAML, ver string, force boo
 	return finalYAML, finalIOPS, nil
 }
 
-func genProfile(helmValues bool, inFilename []string, profile, setOverlayYAML, configPath string, force bool, l *Logger) (string, error) {
-	finalYAML, finalIOPS, err := genIOPS(inFilename, profile, setOverlayYAML, "", force, l)
+func getClusterSpecificValues(config *rest.Config, l *Logger) (string, error) {
+	overlays := []string{
+		getJwtTypeOverlay(config, l),
+	}
+
+	return MakeTreeFromSetList(overlays, false, l)
+
+}
+
+func getJwtTypeOverlay(config *rest.Config, l *Logger) string {
+	d, err := discovery.NewDiscoveryClientForConfig(config)
+	if err != nil {
+		l.logAndErrorf("Failed to determine JWT policy support. Assuming third-party-jwt is supported: %v", err)
```

Comment:
I think the logic should be: jwtPolicy is initialized as first-party-jwt. If the third-party-jwt support is detected, set the jwtPolicy to be third-party-jwt.
The fall-back value of jwtPolicy when failing to determine JWT policy should be first-party-jwt.

Reply:
I feel like if we run into issues we should fall back to the secure option not the insecure option?

Reply:
In this case, the issues are likely to be some platforms do not support third-party-jwt, falling back to first-party-jwt will solve such issues.

Reply:
the errors here are just complete failures to talk to the API server. It seems reasonable to even fail hard here to be honest

Reply:
I agree that it is reasonable to fail hard here if failed to talk to the API server.